eclair-core/src/main/scala/fr/acinq/eclair/channel/publish/TxPublisher.scala

```diff
@@ -220,23 +221,33 @@ private class TxPublisher(nodeParams: NodeParams, factory: TxPublisher.ChildFact
         }
 
       case cmd: PublishReplaceableTx =>
-        val proposedConfirmationTarget = cmd.txInfo.confirmBefore
+        val proposedConfirmationTarget = cmd.txInfo.confirmationTarget
         val attempts = pending.getOrElse(cmd.input, PublishAttempts.empty)
         attempts.replaceableAttempt_opt match {
           case Some(currentAttempt) =>
             if (currentAttempt.cmd.txInfo.tx.txOut.headOption.map(_.publicKeyScript) != cmd.txInfo.tx.txOut.headOption.map(_.publicKeyScript)) {
               log.error("replaceable {} sends to a different address than the previous attempt, this should not happen: proposed={}, previous={}", currentAttempt.cmd.desc, cmd.txInfo, currentAttempt.cmd.txInfo)
             }
-            val currentConfirmationTarget = currentAttempt.confirmBefore
-            if (currentConfirmationTarget <= proposedConfirmationTarget) {
-              log.debug("not publishing replaceable {} spending {}:{} with confirmation target={}, publishing is already in progress with confirmation target={}", cmd.desc, cmd.input.txid, cmd.input.index, proposedConfirmationTarget, currentConfirmationTarget)
-              Behaviors.same
-            } else {
+            val currentConfirmationTarget = currentAttempt.confirmationTarget
+
+            def updateConfirmationTarget() = {
               log.info("replaceable {} spending {}:{} has new confirmation target={} (previous={})", cmd.desc, cmd.input.txid, cmd.input.index, proposedConfirmationTarget, currentConfirmationTarget)
               currentAttempt.actor ! ReplaceableTxPublisher.UpdateConfirmationTarget(proposedConfirmationTarget)
-              val attempts2 = attempts.copy(replaceableAttempt_opt = Some(currentAttempt.copy(confirmBefore = proposedConfirmationTarget)))
+              val attempts2 = attempts.copy(replaceableAttempt_opt = Some(currentAttempt.copy(confirmationTarget = proposedConfirmationTarget)))
               run(pending + (cmd.input -> attempts2), retryNextBlock, channelContext)
             }
+
+            (currentConfirmationTarget, proposedConfirmationTarget) match {
+              case (ConfirmationTarget.Absolute(currentConfirmBefore), ConfirmationTarget.Absolute(proposedConfirmBefore)) if proposedConfirmBefore < currentConfirmBefore =>
+                // The proposed block target is closer than what it was
+                updateConfirmationTarget()
+              case (_: ConfirmationTarget.Priority, ConfirmationTarget.Absolute(proposedConfirmBefore)) =>
+                // Switch from relative priority mode to absolute blockheight mode
+                updateConfirmationTarget()
+              case _ =>
```

Comment:
It could also make sense to handle the case where both targets are using `ConfirmationTarget.Priority` and the new priority is higher (e.g. the node operator has unconfirmed commit txs without htlcs, wants to reclaim their pending liquidity and updates their `closing` priority in `eclair.conf` from `slow` to `medium` or `fast`).

Reply:
I thought about that, but I figured that in this scenario eclair would have been restarted, so there wouldn't be a replacement.

Reply:
That's true, good point!